docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,366 @@
+    with the latter.
+    - The public key of an asymmetric key-pair generated by an attester,
+    serving as an unique public identity of the attester,
+    is a good example of a custom claim.
+  - Custom claims can also be created by a verifier application and passed to
+  an attester for inclusion in the evidence.
+    - This type of claims allows the attester to make assertions about
+    its evidence to the targeted verifier.
+    - A good example of this type of claims is a nonce generated by a
+    verifier application and passed to an attester for inclusion in an
+    evidence, for evidence freshness.
+      - This nonce is generated by the verification application itself,
+      and not by its verifier plugins.
+      - On the attester side, the OE SDK attester plugin simply includes it
+      in the newly-generated evidence, and does not do any interpretation.
+      - Upon successful verification of the evidence, if its nonce is included
+      in the evidence, the verifier application can be sure that the
+      evidence is fresh.
+
+- Optional parameters semantics are the opposite of that of custom claims:
+  - They are TEE and evidence format specific
+  - They are interpreted and processed by OE SDK attester and verifier plugins
+    - Typically, optional parameters are provided by a verifier to an attester,
+    so the attester can generate the correct evidence that can be
+    verified by the verifier
+  - They are not interpreted nor processed by attester / verifier
+  application software
+  - An SGX target info for local  attestation is a good example of
+  optional parameter, as explained in the paragraphs below.
+
+For SGX local attestation, the SGX target info generated by a
+verifier enclave for an attester enclave is a perfect fit for the optional
+parameters in the attester enclave's call to `oe_get_evidence()`:
+- The target info is SGX specific, and is not parsed nor interpreted
+by the TEE-agnostic application code.
+- The SGX target info is generated by the verifier API function
+`oe_verifier_get_format_settings()` for SGX local attestation,
+and passed by the calling verifier application as an opaque blob to an
+attester enclave.
+- The attester enclave invokes its SGX local attestation attester plugin via
+`oe_get_evidence()`, passing in the received SGX target info as an opaque blob,
+to generate an evidence containing the SGX report targeted for the
+verifier enclave.
+- This SGX report can only be verified by the verifier plugin of the verifier
+enclave that generated the SGX target info.
+Any other verifier enclaves, either running on the local or a remote platform,
+won’t be able to verify this SGX report.
+
+## Attester – Verifier Security Model and TEE Agnostic Design
+
+This section explains how the proposed [OE SDK attestation API](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/Attestation_API_Proposal.md#proposed-oe-sdk-v010-public-api)
```

Comment:
```suggestion
This section explains how the proposed [OE SDK attestation API](Attestation_API_Proposal.md#proposed-oe-sdk-v010-public-api)
```
use relative path.  Consider doing the same for other links in the doc.

Reply:
Thanks!